internal/olm/operator/bundle/install.go

```diff
@@ -233,8 +230,10 @@ func addBundleToIndexImage(indexImage string, bundleDeclConfig *declarativeconfi
 		}
 	}
 
-	if bundleNotPresent && len(bundleDeclConfig.Bundles) > 0 && len(bundleDeclConfig.Channels) > 0 {
-		imageDeclConfig.Packages = append(imageDeclConfig.Packages, bundleDeclConfig.Packages[0])
+	if bundleNotPresent {
+		if len(bundleDeclConfig.Packages) > 0 {
+			imageDeclConfig.Packages = append(imageDeclConfig.Packages, bundleDeclConfig.Packages[0])
+		}
```

Comment:
I see that the 2nd and 3rd conditions were moved inside the block. It will _never_ be the case there may be > 0 bundles but 0 channels, right?

I had this condition at the top to avoid the possibility of appending to only one field of the declarative config (i.e., bundle gets appended but not channel, etc.). Not sure if that's possible to begin with.